core/src/main/java/org/polypheny/db/processing/QueryProcessorHelpers.java

```diff
@@ -113,24 +120,36 @@ public static List<ColumnMetaData> getColumnMetaDataList( JavaTypeFactory typeFa
     public static ColumnMetaData.AvaticaType avaticaType( JavaTypeFactory typeFactory, AlgDataType type, AlgDataType fieldType ) {
         final String typeName = type.getPolyType().getTypeName();
         if ( type.getComponentType() != null ) {
-            final ColumnMetaData.AvaticaType componentType = avaticaType( typeFactory, type.getComponentType(), null );
+            ColumnMetaData.AvaticaType componentType = avaticaType( typeFactory, type.getComponentType(), null );
```

Comment:
We should get rid of this Avatica stuff in the core code. Can this be refactored?

Reply:
Not easily, as it is still used throughout, requires second pass.